docs/cart-and-checkout-blocks/custom-inner-checkout-block.md

```diff
@@ -0,0 +1,203 @@
+
+```js
+registerCheckoutBlock( {
+	metadata,
+	component: Block,
+} );
+```
+
+The documentation for `registerCheckoutBlock()` can be found on [github.com](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/packages/checkout/blocks-registry/README.md#registercheckoutblock-options-).
+
+### Control the block visibility and placement
+
+The visibility and placement is controlled by the parent block. For example, if a custom inner block has the parent property `woocommerce/checkout-shipping-address-block`, but the shopper selected local pickup, then the shipping address block will not be visible. As a result of that, the custom inner block will not be visible either.
+
+### Manage the local block state using `useState`
+
+To manage the local block state, we can use the `useState` hook. This allows us to store and update the user input within the block. In our case, we are storing the selected option, e.g. `Try again another day`, and the user input if the shopper selected the option `Other`. This takes place in [the following section](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/1d65e685fb40c207e8e06d6c066b890f527276c5/src/js/shipping-workshop-block/block.js#L58-L62):
+
+```js
+const [
+	selectedAlternateShippingInstruction,
+	setSelectedAlternateShippingInstruction,
+] = useState( 'try-again' );
+const [ otherShippingValue, setOtherShippingValue ] = useState( '' );
+```
+
+The documentation for `useState` can be found on [react.dev](https://react.dev/docs/hooks-state).
+
+### Connect to the checkout lifecycle
+
+We also want to react to certain actions, for example when the shopper selects the option `Other`, which then displays a text field, but does not provide any input. In this case, we want to inform the shopper about the missing input. We are doing this in [src/js/shipping-workshop-block/block.js](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/1d65e685fb40c207e8e06d6c066b890f527276c5/src/js/shipping-workshop-block/block.js#L148-L153):
+
+```js
+setValidationErrors( {
+	[ validationErrorId ]: {
+		message: __( 'Please add some text', 'shipping-workshop' ),
+		hidden: ! hasInteractedWithOtherInput,
+	},
+} );
+```
+
+If the shopper provides a text or selects another option, we want to clear the validation error. This also takes place in [src/js/shipping-workshop-block/block.js](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/1d65e685fb40c207e8e06d6c066b890f527276c5/src/js/shipping-workshop-block/block.js#L139-L147):
+
+```js
+if (
+	selectedAlternateShippingInstruction !== 'other' ||
+	otherShippingValue !== ''
+) {
+	if ( validationError ) {
+		clearValidationError( validationErrorId );
+	}
+	return;
+}
+```
```

Comment:
We need to link to the validation data store docs here, also generalise this point and link off to the rest of the data store docs and any other checkout lifecycle docs we have.

Reply:
Thanks for pointing this out. I'll create a more generic doc and will mention the individual data stores (cart, checkout, validation, etc.) there.